atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -959,6 +961,17 @@ private int estimateSize(List<RowResult<Value>> rangeRows) {
             }
         }
 
+        if (AtlasDbConstants.hiddenTables.contains(tableRef)) {
```

Comment:
Yes, I believe that a better solution would be to set a flag in the table metadata to mark a table as non-transactional, but this also makes it a bit more complicated because now a lot more places would need to have access to the table metadata.